### PR TITLE
feature: adding on-push test workflow

### PR DESCRIPTION
**Pull request by atma-stackoverflow**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: PR Test
+
+on:
+  pull_request:
+    branches: [ master ]
+
```
Comment (MaxHorstmann) [inline .github/workflows/push.yml:6]: Also run on merge?

```
on:
  pull_request:
    branches: [ master ]
  push:
    branches: [ master ]
```

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: PR Test
```
Comment (MaxHorstmann) [inline .github/workflows/push.yml:1]: Name the file `build.yml` for consistency? 

https://github.com/StackExchange/dnscontrol/blob/master/.github/workflows/build.yml

**Review by MaxHorstmann — APPROVED** [review]
👍

File: .github/workflows/push.yml (hunk 1 above)
Comment (atma-stackoverflow) [reply .github/workflows/push.yml:6]: Nope just PR, that's how we do it on all the other things.  Saves resources.

File: .github/workflows/push.yml (hunk 2 above)
Comment (atma-stackoverflow) [reply .github/workflows/push.yml:1]: This is the name of the file we use in the rest of the work I've done.  I also have a release.yml that will be in a future PR and is used for cutting releases.

So this is consistent with the rest of the work I've done.

File: .github/workflows/push.yml (hunk 1 above)
Comment (MaxHorstmann) [reply .github/workflows/push.yml:6]: Running it again on master will catch a bad merge. Build minutes are free on public.

**Pull request merged**